File: GanttChartLightLibraryDemos/Demos/Samples/WPF-CSharp/NetworkDiagramView/ShapeTemplating/CustomNetworkDiagramItem.cs
using System;
using DlhSoft.Windows.Controls.Pert;
using System.ComponentModel;

namespace Demos.WPF.CSharp.NetworkDiagramView.ShapeTemplating
{
    class CustomNetworkDiagramItem : NetworkDiagramItem, INotifyPropertyChanged
    {
        private DateTime actualStart = DateTime.Today;
        public DateTime ActualStart
        {
            get { return actualStart; }
            set
            {
                if (value > ActualFinish)
                    ActualFinish = value;
                actualStart = value;
                OnPropertyChanged("ActualStart");
            }
        }

        private DateTime actualFinish = DateTime.Today;
        public DateTime ActualFinish
        {
            get { return actualFinish; }
            set
            {
                if (value < ActualStart)
                    ActualStart = value;
                actualFinish = value;
                OnPropertyChanged("ActualFinish");
            }
        }
    }
}

File: GanttChartLightLibraryDemos/Demos/Samples.Resources/WPF-VisualBasic/NetworkDiagramView/ShapeTemplating/CustomNetworkDiagramItem.vb
Imports DlhSoft.Windows.Controls.Pert
Imports System.ComponentModel

Friend Class CustomNetworkDiagramItem
    Inherits NetworkDiagramItem
    Implements INotifyPropertyChanged

    Private actualStartValue As Date = Date.Today
    Public Property ActualStart() As Date
        Get
            Return actualStartValue
        End Get
        Set(value As Date)
            If value > ActualFinish Then
                ActualFinish = value
            End If
            actualStartValue = value
            OnPropertyChanged("ActualStart")
        End Set
    End Property

    Private actualFinishValue As Date = Date.Today
    Public Property ActualFinish() As Date
        Get
            Return actualFinishValue
        End Get
        Set(value As Date)
            If value < ActualStart Then
                ActualStart = value
            End If
            actualFinishValue = value
            OnPropertyChanged("ActualFinish")
        End Set
    End Property
End Class
